### PR TITLE
Add 'arn' column to aws_config_configuration_recorder table closes #370

### LINKED ISSUE #370: Add 'arn' column to aws_config_configuration_recorder table
**Is your feature request related to a problem? Please describe.**
I would like to be able to get the ARN of a config recorder quickly without needing to join account/region/partition information together.

**Describe the solution you'd like**
Add an arn column into the table.

**Describe alternatives you've considered**
Joining all of the required information together.

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
SETUP: tests/aws_config_configuration_recorder []
PRETEST: tests/aws_config_configuration_recorder
TEST: tests/aws_config_configuration_recorder
Running terraform
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_region.alternate: Refreshing state...
data.null_data_source.resource: Refreshing state...
data.template_file.resource_aka: Refreshing state...
aws_iam_role.r: Creating...
aws_iam_role.r: Creation complete after 5s [id=turbottest91142]
aws_config_configuration_recorder.named_test_resource: Creating...
aws_config_configuration_recorder.named_test_resource: Creation complete after 2s [id=turbottest91142]
Warning: Deprecated Resource
The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
Outputs:
account_id = 986325076436
aws_partition = aws
region_name = us-west-2
resource_aka = arn:aws:config:us-west-2:986325076436:config-recorder/turbottest91142
resource_name = turbottest91142
Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:config:us-west-2:986325076436:config-recorder/turbottest91142"
    ],
    "arn": "arn:aws:config:us-west-2:986325076436:config-recorder/turbottest91142",
    "name": "turbottest91142",
    "role_arn": "arn:aws:iam::986325076436:role/turbottest91142",
    "status": {
      "LastErrorCode": null,
      "LastErrorMessage": null,
      "LastStartTime": null,
      "LastStatus": null,
      "LastStatusChangeTime": null,
      "LastStopTime": null,
      "Name": "turbottest91142",
      "Recording": false
    },
    "status_recording": false,
    "title": "turbottest91142"
  }
]
✔ PASSED
Running SQL query: test-hydrate-query.sql
[
  {
    "name": "turbottest91142",
    "role_arn": "arn:aws:iam::986325076436:role/turbottest91142",
    "status": {
      "LastErrorCode": null,
      "LastErrorMessage": null,
      "LastStartTime": null,
      "LastStatus": null,
      "LastStatusChangeTime": null,
      "LastStopTime": null,
      "Name": "turbottest91142",
      "Recording": false
    },
    "status_recording": false
  }
]
✔ PASSED
Running SQL query: test-list-call-query.sql
[
  {
    "name": "turbottest91142",
    "role_arn": "arn:aws:iam::986325076436:role/turbottest91142",
    "status": {
      "LastErrorCode": null,
      "LastErrorMessage": null,
      "LastStartTime": null,
      "LastStatus": null,
      "LastStatusChangeTime": null,
      "LastStopTime": null,
      "Name": "turbottest91142",
      "Recording": false
    },
    "title": "turbottest91142"
  }
]
✔ PASSED
Running SQL query: test-not-found-query.sql
null
✔ PASSED
Running SQL query: test-turbot-query.sql
[
  {
    "account_id": "986325076436",
    "akas": [
      "arn:aws:config:us-west-2:986325076436:config-recorder/turbottest91142"
    ],
    "name": "turbottest91142",
    "region": "us-west-2",
    "title": "turbottest91142"
  }
]
✔ PASSED
POSTTEST: tests/aws_config_configuration_recorder
TEARDOWN: tests/aws_config_configuration_recorder
SUMMARY:
1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select name,arn,recording_group,role_arn from aws_config_configuration_recorder
+---------+---------------------------------------------------------------+----------------------------------------------------------------------------+-----------------------------------------------------+
| name    | arn                                                           | recording_group                                                            | role_arn                                            |
+---------+---------------------------------------------------------------+----------------------------------------------------------------------------+-----------------------------------------------------+
| default | arn:aws:config:us-east-1:986325076436:config-recorder/default | {"AllSupported":true,"IncludeGlobalResourceTypes":true,"ResourceTypes":[]} | arn:aws:iam::986325076436:role/turbot/turbot_config |
+---------+---------------------------------------------------------------+----------------------------------------------------------------------------+-----------------------------------------------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_config_configuration_recorder.go [hunk 1]
```diff
@@ -170,17 +177,15 @@ func getConfigConfigurationRecorderStatus(ctx context.Context, d *plugin.QueryDa
 	return status.ConfigurationRecordersStatus[0], nil
 }
 
-//// TRANSFORM FUNCTIONS
```
Comment (cbruno10) [inline aws/table_aws_config_configuration_recorder.go:173]: Why did you remove this comment?

File: aws/table_aws_config_configuration_recorder.go [hunk 2]
```diff
@@ -30,6 +30,13 @@ func tableAwsConfigConfigurationRecorder(_ context.Context) *plugin.Table {
 				Type:        proto.ColumnType_STRING,
 				Transform:   transform.FromField("Name"),
 			},
+			{
+				Name:        "arn",
+				Description: "Amazon resource name of Configuration Recorder",
```
Comment (cbruno10) [inline aws/table_aws_config_configuration_recorder.go:35]: Is this comment consistent with the other `arn` columns? It looks a bit different.

Also, please note that all descriptions should end with periods. Please review description standards here - https://steampipe.io/docs/develop/standards#table-and-column-descriptions

File: aws-test/tests/aws_config_configuration_recorder/variables.tf [hunk 3]
```diff
@@ -7,13 +7,13 @@ variable "resource_name" {
 
 variable "aws_profile" {
   type        = string
-  default     = "integration-tests"
+  default     = "default"
```
Comment (cbruno10) [inline aws-test/tests/aws_config_configuration_recorder/variables.tf]: Did we need to change this? What do other tests use?

File: aws-test/tests/aws_config_configuration_recorder/variables.tf [hunk 4]
```diff
@@ -7,13 +7,13 @@ variable "resource_name" {
...
   description = "AWS credentials profile used for the test. Default is to use the default profile."
 }
 
 variable "aws_region" {
   type        = string
-  default     = "us-east-1"
+  default     = "us-west-2"
```
Comment (cbruno10) [inline aws-test/tests/aws_config_configuration_recorder/variables.tf]: Did we need to change this? What do other tests use?

File: aws-test/tests/aws_config_configuration_recorder/test-hydrate-expected.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "name": "{{resourceName}}",
-    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/awsconfig-example",
+    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{resourceName}}",
```
Comment (cbruno10) [inline aws-test/tests/aws_config_configuration_recorder/test-hydrate-expected.json:4]:
```suggestion
    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{ resourceName }}",
```

File: aws-test/tests/aws_config_configuration_recorder/test-list-call-expected.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "name": "{{resourceName}}",
-    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/awsconfig-example",
+    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{resourceName}}",
```
Comment (cbruno10) [inline aws-test/tests/aws_config_configuration_recorder/test-list-call-expected.json:4]:
```suggestion
    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{ resourceName }}",
```

File: aws-test/tests/aws_config_configuration_recorder/test-get-expected.json [hunk 7]
```diff
@@ -3,8 +3,9 @@
     "akas": [
       "arn:{{ output.aws_partition.value }}:config:{{ output.region_name.value }}:{{ output.account_id.value }}:config-recorder/{{resourceName}}"
     ],
+    "arn": "arn:{{ output.aws_partition.value }}:config:{{ output.region_name.value }}:{{ output.account_id.value }}:config-recorder/{{resourceName}}",
     "name": "{{resourceName}}",
-    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/awsconfig-example",
+    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{resourceName}}",
```
Comment (cbruno10) [inline aws-test/tests/aws_config_configuration_recorder/test-get-expected.json:8]:
```suggestion
    "role_arn": "arn:{{ output.aws_partition.value }}:iam::{{ output.account_id.value }}:role/{{ resourceName }}",
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
Please see comments, thanks!

File: aws-test/tests/aws_config_configuration_recorder/variables.tf (hunk 3 above)
Comment (ParthaI) [reply aws-test/tests/aws_config_configuration_recorder/variables.tf]: Ahh, My bad reverted back

File: aws-test/tests/aws_config_configuration_recorder/variables.tf (hunk 4 above)
Comment (ParthaI) [reply aws-test/tests/aws_config_configuration_recorder/variables.tf]: Ahh, My bad reverted back

File: aws/table_aws_config_configuration_recorder.go (hunk 2 above)
Comment (ParthaI) [reply aws/table_aws_config_configuration_recorder.go:35]: Updated the description as per our standard

File: aws/table_aws_config_configuration_recorder.go (hunk 1 above)
Comment (ParthaI) [reply aws/table_aws_config_configuration_recorder.go:173]: Reverted back

**Review by cbruno10 — APPROVED** [review]
